cli/commands/terraform/action.go

```diff
@@ -781,13 +785,22 @@ func filterTerraformEnvVarsFromExtraArgs(terragruntOptions *options.TerragruntOp
 // Convert the given variables to a map of environment variables that will expose those variables to Terraform. The
 // keys will be of the format TF_VAR_xxx and the values will be converted to JSON, which Terraform knows how to read
 // natively.
-func ToTerraformEnvVars(vars map[string]interface{}) (map[string]string, error) {
+func ToTerraformEnvVars(opts *options.TerragruntOptions, vars map[string]interface{}) (map[string]string, error) {
+	if useLegacyNullValues() {
+		opts.Logger.Warnf("%s is a temporary workaround to bypass the breaking change in #2663.", useLegacyNullValuesEnvVar)
```

Comment:
NIT: It might be better to group this into one logging message for users with JSON logging.